Update readme (#198)
hashicorp/terraform-aws-consul-ecs: README.md

## Architecture

> ![Architecture](./_docs/architecture.png?raw=true)

Reviewer: In order for this README to show up correctly on the [Terraform registry](https://registry.terraform.io/modules/hashicorp/consul-ecs/aws/latest#architecture), this must be an absolute URL. We need to revert this change.

Reviewer: We should take this opportunity to update our docs link to use the https://developer.hashicorp.com/consul/docs/ecs URL. See https://github.com/hashicorp/terraform-aws-consul-ecs/pull/198/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R8

Author: I just used relative paths to make sure the updated images (present in the feature branch) appear properly when rendered as markdown. I have an action to modify the same (at multiple other places as well) when I raise a giant PR against the main branch. WDYT?

Each task is created via the `mesh-task` module. This module adds
additional containers known as sidecar containers to your task definition.

Specifically, it adds the following containers:

> * `consul-ecs-control-plane` – Runs for the full lifecycle of the task. At startup, it connects to the available Consul servers and performs a login with the configured IAM Auth method and obtains an ACL token with appropriate privileges. Using the token, it registers the service and proxy entities to Consul's catalog. It then bootstraps the configuration JSON required by the Consul dataplane container and writes it to a shared volume. After this, the container enters into its reconciliation loop where it periodically syncs the health of ECS containers into Consul. Upon receiving SIGTERM, it marks the corresponding service instance in Consul as unhealthy and waits for the dataplane container to shutdown after which it deregisters the service and proxy entities from Consul's catalog and performs a Consul logout.

Reviewer: I found this block of text kind of hard to grok. What do you think about using sub-bullets to organize it?

```suggestion
* `consul-ecs-control-plane` – Runs for the full lifecycle of the task.
  * At startup it connects to the available Consul servers and performs a login with the configured IAM Auth method to obtain an ACL token with appropriate privileges.
  * Using the token, it registers the service and proxy entities to Consul's catalog.
  * It then bootstraps the configuration JSON required by the Consul dataplane container and writes it to a shared volume.
  * After this, the container enters into its reconciliation loop where it periodically syncs the health of ECS containers into Consul.
  * Upon receiving SIGTERM, it marks the corresponding service instance in Consul as unhealthy and waits for the dataplane container to shutdown.
  * Finally, it deregisters the service and proxy entities from Consul's catalog and performs a Consul logout.
```

* `consul-dataplane` – Runs for the full lifecycle of the task. This container runs

>   the [Consul dataplane](https://github.com/hashicorp/consul-dataplane) binary which is used to configure and start Envoy. Envoy acts as a proxy and controls all the service mesh traffic. All requests to and from the application run through

Reviewer: suggesting some wording updates:

```suggestion
  the [Consul dataplane](https://github.com/hashicorp/consul-dataplane) binary that configures and starts the Envoy proxy, which controls all the service mesh traffic. All requests to and from the application run through
```
